### PR TITLE
libscrypt: fix build failure on Sonoma

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Let's have a closer look at what's happening here...

### CODE REVIEW

File: Formula/lib/libscrypt.rb [hunk 1]
```diff
@@ -17,16 +17,25 @@ class Libscrypt < Formula
   end
 
   def install
+    # `-Os` leads to bugs. https://github.com/technion/libscrypt/issues/60
+    # TODO: Add `ENV.O2` to `brew`.
+    ENV["HOMEBREW_OPTIMIZATION_LEVEL"] = "O2"
```
Comment (fxcoudert) [inline Formula/lib/libscrypt.rb:22]: We could do `ENV.O1`. I've tested and `-O1` does not trigger the bug.

Comment (MikeMcQuaid) [reply Formula/lib/libscrypt.rb:22]: We should do `ENV.O1`. We don't want to add `ENV.O2` to Homebrew/brew because it's already the default on Linux.

Comment (MikeMcQuaid) [reply Formula/lib/libscrypt.rb:22]: We should also not use `HOMEBREW_OPTIMIZATION_LEVEL` like this in formulae.

**Review by p-linnane — APPROVED** [review]
Awesome work @carlocab & @fxcoudert!

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**